### PR TITLE
[cronet_http] Apply relevant rules with the ProGuard

### LINKED ISSUE #1179: http requests fail with `cronetHttpNoPlay=true` with error `java.lang.ClassNotFoundException`
Hello.

I have an app that works fine with `cronet_http` with the GPS-Cronet. Trying to build the same app with `--dart-define=cronetHttpNoPlay=true` works, but the app is not usable and throws errors on http requests. The error message reads:

```
Exception in Java code called through JNI:
java.lang.ClassNotFoundException: Didn't find class "org/chromium/net/CronetEngine$Builder" on path:
DexPathList[[zip file "/data/app/~~xlxrt1ba2YAQ0amUKmEvNg==/com.dreautall.waterflyiii-AbxaklwneLCpq2CmygW_Jg==/base.apk"],nativeLibrary Directories=[/data/app/~~xlxrt1ba2YAQ0amUKmEvNg==/com.dreautall.waterflyiii-AbxaklwneLCpq2CmygW_Jg==/lib/arm64,/data/app/~~xlxrt1ba2YAQ0amUKmEvNg==/com.dreautall.waterflyiii-AbxaklwneLCpq2CmygW_Jg==/base.apk!/lib/arm64-v8a, /system/lib64,/system/system_ext/lib64]]

java.lang.ClassNotFoundException: Didn't find class "org/chromium/net/CronetEngine$Builder" on path:
DexPDexPathList[[zip file "/data/app/~~xlxrt1ba2YAQ0amUKmEvNg==/com.dreautall.waterflyiii-AbxaklwneLCpq2CmygW_Jg==/base.apk"],nativeLibrary Directories=[/data/app/~~xlxrt1ba2YAQ0amUKmEvNg==/com.dreautall.waterflyiii-AbxaklwneLCpq2CmygW_Jg==/lib/arm64,/data/app/~~xlxrt1ba2YAQ0amUKmEvNg==/com.dreautall.waterflyiii-AbxaklwneLCpq2CmygW_Jg==/base.apk!/lib/arm64-v8a, /system/lib64,/system/system_ext/lib64]]

at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:259)
at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
```

pubspec.yaml defines `cronet_http: ^1.2.0`. As no other change is outlined in the documentation, there was no code change from my side anywhere - not sure if I would have to exclude/include something in the Android manifest, gradle build or similar.

The app was built with `flutter build apk --split-debug-info=build/app/outputs/symbols/aab --release --dart-define=cronetHttpNoPlay=true`, the test device of the error message is a Samsung Galaxy A52 running the stock OS, but it also fails on a Pixel 8.

### PR DESCRIPTION
**Pull request by AlexV525**
Resolves #1179

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

**Comment by AlexV525** [comment]
The failing test looks unrelated. https://github.com/dart-lang/http/actions/runs/9140740181/job/25134408655?pr=1204

File: pkgs/cronet_http/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 1.2.2-wip
```
Comment (brianquinlan) [inline pkgs/cronet_http/CHANGELOG.md:1]: Could you change this to 1.3.1-wip?

File: pkgs/cronet_http/pubspec.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: cronet_http
-version: 1.2.1
+version: 1.2.2-wip
```
Comment (brianquinlan) [inline pkgs/cronet_http/pubspec.yaml:2]: 1.3.1-wip

**Review by brianquinlan — CHANGES_REQUESTED** [review]

**Comment by brianquinlan** [comment]
Thanks @AlexV525 !

**Review by brianquinlan — APPROVED** [review]

**Pull request merged**